x-pack/plugins/enterprise_search/server/routes/enterprise_search/engines.ts

```diff
@@ -171,11 +171,10 @@ export function registerEnginesRoutes({ config, log, router }: RouteDependencies
       const engineName = decodeURIComponent(request.params.engine_name);
       const { client } = (await context.core).elasticsearch;
 
-      const engine = await fetchEnterpriseSearch<EnterpriseSearchEngineDetails>(
-        config,
-        request,
-        `/api/engines/${engineName}`
-      );
+      const engine = await client.asCurrentUser.transport.request<EnterpriseSearchEngine>({
```

Comment:
If a non 2xx response, double check this doesn't throw an exception instead. The pattern i've been using is using is catching the exception and identifying error with the `isNotFoundException` helpers in `identify_exceptions`